spec.md

```diff
@@ -167,6 +167,43 @@ merged are hosted in other folders.
 As some Compose file elements can both be expressed as single strings or complex objects, merges MUST apply to
 the expanded form.
 
+### Profiles
+
+Profiles allow to adjust the Compose application model for various usages and environments. A Compose
+implementation SHOULD allow user to define a set of active profiles. The exact mechanism is implementation
+specific and MAY include command line flags, environment variables, etc.
+
+All top-level elements support a `profiles` attribute to define a list of named profiles. Elements without 
+a `profiles` attribute set MUST always be enabled. The enclosing component MUST be ignored by compose
+implementation when none of the listed profiles match the active ones, until the target component is
+explicitly targeted by a command.
```

Comment:
It feels a bit strange that users can get (what is essentially) a one-off profile with always-on services + just the target service. This might get rather confusing since we're ignoring `profiles` attribute on the target service, but enforcing it everywhere else.

Is it worth to fomalise this a bit more by saying that `compose run/up TARGET` will activate the `profiles` of the target service? I appreciate that this probably makes the lives of Compose implementors harder (especially considering the follow-up profile-specific files proposal), but this functionality in isolation seems more confusing than beneficial.